packages/framework/src/validators/class-validator.validator.ts

```diff
@@ -0,0 +1,139 @@
+import { ValidationError } from 'class-validator';
+import type { ClassType, FromSchema, FromSchemaUnvalidated, JsonSchema, Schema } from '../types/schema.types';
+import type { ValidateResult, Validator } from '../types/validator.types';
+
+// Function to recursively add `additionalProperties: false` to the schema
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function replaceSchemaRefs(schema: any, schemas: any): JsonSchema {
+  if (schema && typeof schema === 'object' && schema?.$ref) {
+    // eslint-disable-next-line no-param-reassign
+    schema = schemas[schema.$ref.split('/').at(-1)];
+  }
+
+  if (schema && typeof schema === 'object')
+    for (const key in schema) {
+      if (schema.hasOwnProperty(key)) {
+        // eslint-disable-next-line no-param-reassign
+        schema[key] = replaceSchemaRefs(schema[key], schemas);
+      }
+    }
+
+  return schema;
+}
+// Function to recursively add `additionalProperties: false` to the schema
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function addAdditionalPropertiesFalse(schema: any): JsonSchema {
+  if (schema && typeof schema === 'object' && schema?.type === 'object') {
+    // eslint-disable-next-line no-param-reassign
+    schema.additionalProperties = false;
+  }
+
+  if (schema.properties) {
+    for (const key in schema.properties) {
+      if (schema.properties.hasOwnProperty(key)) {
+        addAdditionalPropertiesFalse(schema.properties[key]);
+      }
+    }
+  }
+
+  return schema;
+}
+function formatErrors(errors: ValidationError[], parentPath = ''): { path: string; message: string }[] {
+  return errors.flatMap((err) => {
+    const currentPath = `${parentPath}/${err.property}`.replace(/\/+/g, '/');
+
+    if (err.children && err.children.length > 0) {
+      // Recursively format the children
+      return formatErrors(err.children, currentPath);
+    } else {
+      // Base case: no children, return the formatted error
+      return {
+        path: currentPath,
+        message: Object.values(err.constraints || {}).join(', '),
+      };
+    }
+  });
+}
+
+export class ClassValidatorValidator implements Validator<ClassType> {
+  canHandle(schema: Schema): schema is ClassType {
+    return typeof (schema as ClassType) === 'function';
+  }
+
+  async validate<
+    T_Schema extends ClassType = ClassType,
+    T_Unvalidated = FromSchemaUnvalidated<T_Schema>,
+    T_Validated = FromSchema<T_Schema>,
+  >(data: T_Unvalidated, schema: T_Schema): Promise<ValidateResult<T_Validated>> {
+    try {
+      // eslint-disable-next-line global-require
+      require('reflect-metadata') as typeof import('reflect-metadata');
+      // eslint-disable-next-line global-require
+      const { plainToInstance, instanceToPlain } = require('class-transformer') as typeof import('class-transformer');
+      // eslint-disable-next-line global-require
+      const { validate } = require('class-validator') as typeof import('class-validator');
```

Comment:
We can convert all of these to use `import` now 🎉

```suggestion
      await import('reflect-metadata');
      const { plainToInstance, instanceToPlain } = await import('class-transformer');
      const { validate } = await import('class-validator');
```